### DISCUSSION TITLE
Features I believe are needed.

### DISCUSSION THREAD

**Discussion by careercoder** [django-cms/django-cms#7119] [General]
Hi, if I'm missing something or if this already exist please excuse my ignorance as I have looked for these items and have not found them to no aval.  If they are existing please help me by linking.

1.) I believe a standard plugin for menu creation should exist. 
While we have the "Pages" menu, this to me is not sufficient enough to handle a few of the complex task that djangocms has the potential to run.

2.) Every page should at some level, have a 'component' view - to me a component view is the main purpose of the page viewing viewed, in-some cases if only 'plugins' are needed it could be considered a BLANK component. 

I understand we have 'apphook' but apphook for me has not worked for pages that are within pages that need modifications of plugin placement.

**Comment by marksweb** [comment]
Could you expand on what you've said there please as I'm not quite understanding exactly what it is you need?

I've got a fairly comprehensive list of jobs on, amongst which is to build a menu/nav plugin so I can appreciate that request!! 👍 

I'm not entirely sure what you mean by a "component view" however.

**Reply by Itotutona** [reply]
Is this a block view like in wordpress?
